python_modules/dagster/dagster/_core/definitions/asset_checks.py

```diff
@@ -1,201 +1,38 @@
-
-    @public
-    @property
-    def specs(self) -> Iterable[AssetCheckSpec]:
-        return self._specs_by_output_name.values()
-
-    @property
-    def keys(self) -> Iterable[AssetCheckKey]:
-        return self._specs_by_handle.keys()
-
-    @property
-    def specs_by_output_name(self) -> Mapping[str, AssetCheckSpec]:
-        return self._specs_by_output_name
-
-    @property
-    def asset_keys_by_input_name(self) -> Mapping[str, AssetKey]:
-        return self._input_output_props.asset_keys_by_input_name
-
-    def get_resource_requirements(self) -> Iterator[ResourceRequirement]:
-        yield from self.node_def.get_resource_requirements()  # type: ignore[attr-defined]
-        for source_key, resource_def in self._resource_defs.items():
-            yield from resource_def.get_resource_requirements(outer_context=source_key)
-
-    def get_spec_for_check_key(self, asset_check_key: AssetCheckKey) -> AssetCheckSpec:
-        return self._specs_by_handle[asset_check_key]
-
-    @public
-    @property
-    def required_resource_keys(self) -> Set[str]:
-        """Set[str]: The set of keys for resources that must be provided to this AssetsDefinition."""
-        return {requirement.key for requirement in self.get_resource_requirements()}
-
-    @property
-    def resource_defs(self) -> Mapping[str, ResourceDefinition]:
-        return self._resource_defs
-
-    def with_resources(
-        self, resource_defs: Mapping[str, ResourceDefinition]
-    ) -> "AssetChecksDefinition":
-        attributes_dict = self.get_attributes_dict()
-        attributes_dict["resource_defs"] = merge_resource_defs(
-            old_resource_defs=self._resource_defs,
-            resource_defs_to_merge_in=resource_defs,
-            requires_resources=self,
-        )
-        return self.__class__(**attributes_dict)
-
-    def get_attributes_dict(self) -> Dict[str, Any]:
-        return dict(
-            node_def=self._node_def,
-            resource_defs=self._resource_defs,
-            specs=self._specs,
-            input_output_props=self._input_output_props,
-        )
-
-    def with_attributes(
-        self,
-        asset_key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
-    ) -> "AssetChecksDefinition":
-        attributes_dict = self.get_attributes_dict()
-        if asset_key_prefix is not None:
-            attributes_dict["specs"] = [
-                spec.with_asset_key_prefix(asset_key_prefix) for spec in self._specs
-            ]
-            attributes_dict["input_output_props"] = self._input_output_props.with_asset_key_prefix(
-                asset_key_prefix
-            )
 
-        return AssetChecksDefinition(**attributes_dict)
+# This is still needed in a few places where we need to handle normal AssetsDefinition and
+# AssetChecksDefinition differently, but eventually those areas should be refactored and this should
+# be removed.
+def is_asset_checks_def(assets_def: AssetsDefinition) -> bool:
```

Comment:
can we call it `has_only_asset_checks` or something? I think that is more accurate.